js/index.js

```diff
@@ -0,0 +1,325 @@
+
+    if (productid == "" || productname == "" || price == "" || description == "") {
+        alert("All fields are required!");
+        return false;
+    }
+    else if (isNaN(productid)) {
+        alert("Only numeric Value");
+        return false;
+    }
+    else if (price < 0) {
+        alert("Price must be positive.")
+        return false;
+    }
+    else if (!description.match(regex) || !productname.match(regex)) {
+        alert("Enter proper details.");
+        return false;
+    }
+    return true;
+}
+
+//Function to validate Image
+function validateImage() {
+    var image = document.getElementById('image');
+    var file = image.value;
+    var extensions = /(\.jpg|\.jpeg|\.png|\.gif)$/i;
+
+    if (!extensions.exec(file)) {
+        alert('Invalid file type');
+        fileInput.value = '';
+        return false;
+    }
+    return true;
+}
+
+//Function to validate name and Id during add data
+function validateIdName() {
+    var productid = document.getElementById("productid").value;
+    var productname = document.getElementById("productname").value;
+    var productList;
+    productList = (localStorage.getItem("productList") == null) ? productList = [] : JSON.parse(localStorage.getItem("productList"));
+    for (let i = 0; i < productList.length; i++) {
+        if (productList[i].productid === productid || productList[i].productname.toLowerCase() === productname.toLowerCase()) {
+            swal("Enter unique Product Id And Name!", {
+                icon: "warning",
+                timer: 2000
+            });
+            return false;
+        }
+    }
+    return true;
+}
+
+//Function to validate Name while editing data
+function validateName(index) {
+    var productname = document.getElementById("productname").value;
+    var productList;
+    productList = (localStorage.getItem("productList") == null) ? productList = [] : JSON.parse(localStorage.getItem("productList"));
+    for (let i = 0; i < productList.length; i++) {
+        if (productList[i].productname.toLowerCase() === productname.toLowerCase()) {
+            if (productList[index].productname.toLowerCase() === productname.toLowerCase()) {
+                return true;
+            }
+            else {
+                swal("Enter unique Product Name!", {
+                    icon: "warning",
+                    timer: 2000
+                });
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+//Function to clear data after using modal
+function clearValue() {
+    $('#addAdmin').modal('hide');
+    document.getElementById("productid").value = "";
+    document.getElementById("productname").value = "";
+    document.getElementById("image").value = "";
+    document.getElementById("imgedit").src = "";
+    document.getElementById("price").value = "";
+    document.getElementById("description").value = "";
+}
+
+//Function to sort/filter
+function changeBy(x) {
```

Comment:
Define exact name of parameter. Don't use xyz or abcd

Reply:
ok sir changed.